### DISCUSSION TITLE
Make sortDescFirst or "next sorting order" available on Column

### DISCUSSION THREAD

**Discussion by NixBiks** [TanStack/table#3944] [Ideas]
AFAIK it's not possible to know what the sorting order will be on next click. It is being determined by

```javascript
          const sortDescFirst =
            column.columnDef.sortDescFirst ??
            instance.options.sortDescFirst ??
            column.getAutoSortDir() === 'desc'
```

from [here](https://github.com/TanStack/react-table/blob/184d8a29aa6a71ad26307e61fdea956adf1825ef/packages/table-core/src/features/Sorting.ts#L222).

#### Use case

Show sorting order on hover, e.g. see this [example](https://tailwindui.com/components/application-ui/lists/tables#component-7b5a46e74e475708d966ca31716f1771)

#### Current solution

Currently I simply mimick the above to determine the initial sort order for each column - it just feels a bit like hacking into the internals

**Comment by rodolphoasb** [comment]
Hey, @mr-bjerre! I guess your use case can be handled by using `header.column.getIsSorted()`

**Reply by NixBiks** [reply]
If that's false then you don't know if it will be ascending or descending next. That's determined by the code I put above AFAIK

**Reply by rodolphoasb** [reply]
This is not the behavior that I'm seeing when I use `header.column.getToggleSortingHandler()`. 

For me this is the current behavior when I click:

No sorting -> 'asc' -> 'desc' -> No sorting.

You can also see this in this [example](https://codesandbox.io/s/github/TanStack/react-table/tree/alpha/examples/react/sorting).

That way, getting the sorting order of the next click is possible.

**Reply by NixBiks** [reply]
> This is not the behavior that I'm seeing when I use `header.column.getToggleSortingHandler()`. 
> 
> 
> 
> For me this is the current behavior when I click:
> 
> 
> 
> No sorting -> 'asc' -> 'desc' -> No sorting.
> 
> 
> 
> You can also see this in this [example](https://codesandbox.io/s/github/TanStack/react-table/tree/alpha/examples/react/sorting).
> 
> 
> 
> That way, getting the sorting order of the next click is possible.

Have you tried with both numbers and non-numbers columns? They have different defaults

**Reply by rodolphoasb** [reply]
Yeah, you're right with numbers the sort order is inversed... 

Well, I guess you could convert all your data cells to strings. But IMO, I also feel like it's strange the starting sort order to be different according to the data type. Maybe open an issue?!

**Reply by NixBiks** [reply]
Well I did solve it properly - I'm just making this thread as a feature suggestion; I.e. to expose the "next sort order" on each column.

**Comment by KevinVandy** [comment]
I like this idea and would use something like `column.getNextSortingOrder()` myself. Propose a PR?

**Reply by NixBiks** [reply]
> I like this idea and would use something like `column.getNextSortingOrder()` myself. Propose a PR?

I might give it a shot later this week

**Reply by NixBiks** [reply]
Made a start in #3956
